explainer.md

```diff
@@ -186,225 +181,236 @@ https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b
+  - The window is top-level, although nested page transitions may be supported in future.
+- `event.transition.setElement(element, tag, options)` - Set an element to be used in the transition.
+  - `element` - the element.
+  - `tag` - the tag name. Can be z to un-set this element.
+  - `options` - reserved for future use. This is where "computed style + content image" and "retaining hierarchy" modes will be exposed.
+- `event.transition.setData(data)` - An object that is structured-cloned and passed to the next page.
+- `event.transition.ignoreCSSTaggedElements()`
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/77): Is this a tight-coupling with app-history, or could it be usable without it?
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/78): Do we need `offerElement`? The same thing could be done by adding the CSS properties.
+Methods on `event.transition` must be called during the dispatch of the `pagehide` event, otherwise an error is thrown.
 
-```js
-// In Page-B
-document.performTransition((transition) => {
-  console.log(transition.data);
-  const pageAHeader = transition.offeredElements.get('header');
-  const pageBHeader = transition.createTransitionElement(document.querySelector('.header'), {
-    capture: 'container-and-child',
-  });
-  transition.matchElements(pageAHeader, pageBHeader);
-});
-```
+Once `pagehide` has dispatched:
 
-This sketch is particularly half-baked. A more concrete proposal will be possible when more of the concepts are decided.
+1. If `ignoreCSSTaggedElements` was not called, gather elements offered via `page-transition-tag`.
+1. Add/remove offered elements according to `setElement` calls.
+1. If at least one element remains offered, a transition can go ahead.
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/79): Do we need `createTransitionElement`? It could be done via adding CSS properties, but it might be clumsy if the developer is going to immediately remove the properties afterwards.
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/80): How does the outgoing page offer just the root for transition?
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/81): When is the `performTransition` callback called?
+As a result of the above, if an element is offered via CSS and `setElement`, then `setElement` wins. If there are multiple calls to `setElement` for the same element, the last wins.
 
-## Defining the animation
+Gathering offered elements _after_ the dispatch of `pagehide` means the developer can use a mix of the CSS and JS methods. For instance, the developer could set a class on the root element depending on `event.nextUrl`, which changes the offered elements.
 
-How will Page-B define the animation?
+## Page-B
 
-### Default animation
+Page-B must also offer elements to use for a transition to happen. Page-B also controls the animation of the transition.
```

Comment:
Both of the "musts" here preclude exit/enter transitions. Is that intentional?

Reply:
I don't think it does. Page-B must offer one element for the transition. What would it look like if it offered none? It would only be Page-A things fading out to… what? A white background?